models/news.model.js

```diff
@@ -59,3 +59,20 @@ exports.articleDataByDate = () => {
       return article;
     });
 };
+
+exports.commentsById = (id) => {
+  return db
+    .query(
+      `SELECT articles.article_id, comments.* FROM articles
```

Comment:
This isn't necessary for commentsById! But it is needed for add comment count to article. We just need to select all from the comments where the article_id matches the one passed. It is also here that we should check the article exists before checking the comments for it.

Reply:
I think this checks if it exists before looking for the comments unless there is specific code to be in there. I looked in the notes in complex queries but it is also talking about WHERE .... which i have in the query anyway. Let me know if I need to add something.

Reply:
So I see what you're getting at with this, but we need to consider how will it handle the difference between an article that doesn't exist VS an article that exists but has no comments. if the article exists but has no comments, our rows on line 72 will be returned as an empty array. However, this will also be the case if we ask SQL to go in and select any comments where that article_id matches. It won't error out - because we haven't done any bad sql - but nor will the array of rows returned have anything inside it. We have to tell our code how to tell the difference.